### DISCUSSION TITLE
Why should renovate CI logs be private?

### DISCUSSION THREAD

**Discussion by Yannik** [renovatebot/renovate#30033] [Request Help]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, latest

### Please tell us more about your question or problem

I noticed the following paragraph in the docs:
> We recommend you use a new and dedicated private project to host the Renovate runner, however a public project with private CI logs should still be safe.

This would suggest that the approach of [having renovate directly in the projects CI](https://github.com/renovatebot/renovate/discussions/30032) would not be advisable (since the CI logs are of course visible to all project members). What is the reason for requiring the CI logs to be private?

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by HonkingGoose** [comment]
The [Renovate docs, Security and Permissions](https://docs.renovatebot.com/security-and-permissions/) explains in more general terms:

- Renovate's security stance
- The Permissions that Renovate needs to do work
- Privacy
- Notes for Hosted/Saas (the Mend Renovate App) users

So that is probably good to read. :wink: 

You already found the [Renovate docs about GitLab's security model](https://docs.renovatebot.com/gitlab-bot-security/), so this link is just for others to be on the same page as you. :wink: 

I found this discussion, which seems a bit related, _maybe_:

- #29898

I'll let a maintainer respond to this part:

> This would suggest that the approach of https://github.com/renovatebot/renovate/discussions/30032 would not be advisable (since the CI logs are of course visible to all project members). What is the reason for requiring the CI logs to be private?

**Comment by RahulGautamSingh** [comment]
To protect sensitive information eg. private registry URLs (or anything else) since not all the fields are masked in the logs. I would assume that project members can be trusted with that kind of info, else you can only be the project member.

---

Edited by @HonkingGoose to fix up some typos.
